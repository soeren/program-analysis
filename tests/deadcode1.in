// Check for dead code elimination
// - Check that dead assignments are removed
// - Check that liveness propagation is correct in/after IF/WHILE
// statements
// - Check that unused defines are removed

program

int y;
int x;
int z;
int q;
int w; // dead

y := 1; // faint
x := 3;
y := y * x; // dead
q := 42;
z := 0;

while (x < 6) {
    x := x + 1;
    z := y * y; // dead bcs of IF
    if ( x > q)
        z := 0;
    else
        z := 1;
}

y := z * q;
if(z > 0){
    y := 0;
} else {
    x := 0;
}
write(y*x);
end
