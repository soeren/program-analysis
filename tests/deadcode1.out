program
int y;
int x;
int z;
int q;
x := 3; // 1
q := 42; // 2
z := 0; // 3
while(x < 6) // 4
{
     x := x + 1; // 5
     if(x > q) // 6
     {
          z := 0; // 7
      } else
     {
          z := 1; // 8
      }
 }
y := z * q; // 9
if(z > 0) // 10
{
     y := 0; // 11
 } else
{
     x := 0; // 12
 }
write(y * x); // 13
end
